state/addmachine.go

```diff
@@ -597,12 +602,14 @@ func (st *State) machineStorageOps(
 		}
 		volumeOps = append(volumeOps, ops...)
 		volumeAttachments = append(volumeAttachments, volumeAttachmentTemplate{
-			tag, v.Attachment,
+			tag, v.Attachment, false,
+		})
+	}
+	for tag, volumeAttachment := range args.volumeAttachments {
+		volumeAttachments = append(volumeAttachments, volumeAttachmentTemplate{
+			tag, volumeAttachment, true,
```

Comment:
would be nice to have a const ExistingStorage=true and NewStorage=false or something and use the consts instead of true/false